c++/src/kj/common-test.c++

```diff
@@ -818,6 +818,188 @@ KJ_TEST("ArrayPtr operator ==") {
+  KJ_EXPECT((ArrayPtr<const double>({0.0}) == ArrayPtr<const double>({0.0})));
+  KJ_EXPECT(!(ArrayPtr<const double>({0.0}) != ArrayPtr<const double>({0.0})));
+  KJ_EXPECT((ArrayPtr<const double>({0.0}) <= ArrayPtr<const double>({0.0})));
+  KJ_EXPECT((ArrayPtr<const double>({0.0}) >= ArrayPtr<const double>({0.0})));
+  KJ_EXPECT(!(ArrayPtr<const double>({0.0}) > ArrayPtr<const double>({0.0})));
+  KJ_EXPECT(!(ArrayPtr<const double>({0.0}) < ArrayPtr<const double>({0.0})));
+  KJ_EXPECT((ArrayPtr<const double>({1.0}) <=> ArrayPtr<const double>({0.0})) == std::partial_ordering::greater);
+  KJ_EXPECT(!(ArrayPtr<const double>({1.0}) == ArrayPtr<const double>({0.0})));
+  KJ_EXPECT((ArrayPtr<const double>({1.0}) != ArrayPtr<const double>({0.0})));
+  KJ_EXPECT(!(ArrayPtr<const double>({1.0}) <= ArrayPtr<const double>({0.0})));
+  KJ_EXPECT((ArrayPtr<const double>({1.0}) >= ArrayPtr<const double>({0.0})));
+  KJ_EXPECT(!(ArrayPtr<const double>({1.0}) < ArrayPtr<const double>({0.0})));
+  KJ_EXPECT((ArrayPtr<const double>({1.0}) > ArrayPtr<const double>({0.0})));
+  KJ_EXPECT((ArrayPtr<const double>({0.0}) <=> ArrayPtr<const double>({1.0})) == std::partial_ordering::less);
+  KJ_EXPECT(!(ArrayPtr<const double>({0.0}) == ArrayPtr<const double>({1.0})));
+  KJ_EXPECT((ArrayPtr<const double>({0.0}) != ArrayPtr<const double>({1.0})));
+  KJ_EXPECT((ArrayPtr<const double>({0.0}) <= ArrayPtr<const double>({1.0})));
+  KJ_EXPECT(!(ArrayPtr<const double>({0.0}) >= ArrayPtr<const double>({1.0})));
+  KJ_EXPECT((ArrayPtr<const double>({0.0}) < ArrayPtr<const double>({1.0})));
+  KJ_EXPECT(!(ArrayPtr<const double>({0.0}) > ArrayPtr<const double>({1.0})));
+  KJ_EXPECT((ArrayPtr<const double>({0.0, 0.0}) <=> ArrayPtr<const double>({0.0})) == std::partial_ordering::greater);
+  KJ_EXPECT(!(ArrayPtr<const double>({0.0, 0.0}) == ArrayPtr<const double>({0.0})));
+  KJ_EXPECT((ArrayPtr<const double>({0.0, 0.0}) != ArrayPtr<const double>({0.0})));
+  KJ_EXPECT(!(ArrayPtr<const double>({0.0, 0.0}) <= ArrayPtr<const double>({0.0})));
+  KJ_EXPECT((ArrayPtr<const double>({0.0, 0.0}) >= ArrayPtr<const double>({0.0})));
+  KJ_EXPECT(!(ArrayPtr<const double>({0.0, 0.0}) < ArrayPtr<const double>({0.0})));
+  KJ_EXPECT((ArrayPtr<const double>({0.0, 0.0}) > ArrayPtr<const double>({0.0})));
+
+  double d[1] = { nan() };
+  KJ_EXPECT((ArrayPtr<double>(d, 1) <=> ArrayPtr<double>(d, 1)) == std::partial_ordering::unordered);
+  KJ_EXPECT(!(ArrayPtr<double>(d, 1) == ArrayPtr<double>(d, 1)));
+  KJ_EXPECT((ArrayPtr<double>(d, 1) != ArrayPtr<double>(d, 1)));
+  KJ_EXPECT(!(ArrayPtr<double>(d, 1) <= ArrayPtr<double>(d, 1)));
+  KJ_EXPECT(!(ArrayPtr<double>(d, 1) >= ArrayPtr<double>(d, 1)));
+  KJ_EXPECT(!(ArrayPtr<double>(d, 1) < ArrayPtr<double>(d, 1)));
+  KJ_EXPECT(!(ArrayPtr<double>(d, 1) > ArrayPtr<double>(d, 1)));
+  // operator== should not use memcmp for double elements.
+}
```

Comment:
let's add test for a structs with custom operators. I think we need to check cases like: struct with <=> only, struct with == only, maybe couple other examples (if any) from our code

Reply:
is the goal to figure out the minimum generators for operations?